include/nbl/builtin/hlsl/bxdf/ndf.hlsl

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2018-2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_BXDF_NDF_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BXDF_NDF_INCLUDED_
+
+#include "nbl/builtin/hlsl/limits.hlsl"
+#include "nbl/builtin/hlsl/bxdf/common.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace bxdf
+{
+namespace ndf
+{
+
+// common
+namespace impl
+{
+template<typename T, bool ggx = false>
```

Comment:
instead of having a bool, you can define a trait `ndf::is_ggx_v<NDF>` and take the `NDF` instead of the `T`

This way you get a more "rock solid" partial spec

Reply:
that will require you to move the `microfaced_to_light_measure_transform` to the bottom of the file